### ISSUE TITLE
Sonarr importing multiple episodes simultaneously

### ISSUE THREAD

**Issue by enz1ey** [mdhiggins/sickbeard_mp4_automator#593]
I use this script in conjunction with acd_cli on a VPS with very limited storage, so my download process is as follows:

1.  Sonarr sends release to SabNZBD
2. SabNZBD downloads release, calls SABPostProcess.py
3. Video file(s) are converted to MP4, then Sonarr is called to scan folder
4. SabNZBD sees post-processing as finished and download "complete" as soon as Sonarr is called to scan

The problem is, since SabNZBD is seeing the download as "complete" while Sonarr is still moving/renaming the file(s), it moves onto the next download. Even if I have SabNZBD pause downloads while post-processing, it still downloads faster than Sonarr can move the files. I have another script called by Sonarr which clears out the local target directory to where episodes are moved after uploading to ACD, but the problem is when it is called for more than one episode at once, it will still be in the process of moving the second episode, but it will delete the folder structure of the directory after the first episode is moved.

Basically, for some reason Sonarr will allow the SabPostProcess.py script to alert it multiple times to scan a folder, and it won't queue those requests. It'll start moving multiple files if called multiple times. On the other hand, when using the same scripts with CouchPotato, SabNZBD will wait until that file is completely moved and renamed by CP before marking the download as "complete" and moving onto the next one. It won't do that for Sonarr.

Is there a way to "slow" Sonarr down and limit it to one file at a time?

**Comment by mdhiggins** [comment]
That's more of a question for the Sonarr people than me, I'm honestly not sure. Either way though slowing it down would not be ideal. Sonarr should only be getting told about subdirectories anyway and should be capable for cleaning up the source files / download directories as it finishes accepting the file. You shouldn't need a custom script to do this and that may be why you're having issues.

If you use Sonarr's built in clean up features it will only delete the files associated with the download its finished, it wont just blanket delete your entire SAB downloads folder.
